### PR TITLE
DDCE-2274:Scala 2.12 Upgrade

### PR DESCRIPTION
**Pull request by manpreetsingh1000**

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -1,24 +1,14 @@
...
+  .enablePlugins(GatlingPlugin)
+  .settings(
+    name := "digital-tariffs-trader-performance-tests",
+    version := "0.1.0-SNAPSHOT",
+    scalaVersion := "2.12.14",
+    //implicitConversions & postfixOps are Gatling recommended -language settings
+    scalacOptions ++= Seq("-feature", "-language:implicitConversions", "-language:postfixOps"),
+    // Enabling sbt-auto-build plugin provides DefaultBuildSettings with default `testOptions` from `sbt-settings` plugin.
+    // These testOptions are not compatible with `sbt gatling:test`. So we have to override testOptions here.
+    testOptions in Test := Seq.empty,
```
Comment (JMaio) [inline build.sbt:11]: should this include something like `parallelExecution in Test := false`?

File: project/plugins.sbt [hunk 2]
```diff
@@ -1,6 +1,7 @@
 resolvers += "HMRC-open-artefacts-maven" at "https://open.artefacts.tax.service.gov.uk/maven2"
 resolvers += Resolver.url("HMRC-open-artefacts-ivy", url("https://open.artefacts.tax.service.gov.uk/ivy2"))(Resolver.ivyStylePatterns)
 
-addSbtPlugin("io.gatling" % "gatling-sbt" % "2.1.7")
-addSbtPlugin("uk.gov.hmrc" % "sbt-settings" % "3.9.0")
-addSbtPlugin("uk.gov.hmrc" % "sbt-auto-build" % "2.15.0")
+addSbtPlugin("io.gatling" % "gatling-sbt" % "3.2.1")
+addSbtPlugin("uk.gov.hmrc" % "sbt-settings" % "4.9.0")
+addSbtPlugin("uk.gov.hmrc" % "sbt-auto-build" % "2.10.0")
```
Comment (JMaio) [inline project/plugins.sbt]: why revert to `2.10.0`?

File: build.sbt (hunk 1 above)
Comment (manpreetsingh1000) [reply build.sbt:11]: The confluence guide says to : Remove the setting parallelExecution in Test - as this comes from sbt-settings via SbtAutoBuildPlugin.

**Review by JMaio — APPROVED** [review]
changes look good, smoke test runs okay, Jenkins run passing:
https://performance.tools.staging.tax.service.gov.uk/view/Digital%20Tariffs/job/digital-tariffs-trader-performance-tests/14/

**Pull request merged**